test/e2e/performanceprofile/functests/1_performance/irqbalance.go

```diff
@@ -301,8 +302,190 @@ var _ = Describe("[performance] Checking IRQBalance settings", Ordered, func() {
+		return false, nil
+	}
+
+	// There are CPUs in minus set, i.e., assigned cpus which are not housekeeping.
+	// All of housekeeping must still be in SMP, but also, minus set must NOT be in SMP (should not be subset).
+	isHousekeepingSubset := housekeepCPUSet.IsSubsetOf(irqSmpAffinity)
+	isMinusSetSubset := minusCpuSet.IsSubsetOf(irqSmpAffinity)
+
+	if isHousekeepingSubset && !isMinusSetSubset {
+		return true, nil
+	}
+
+	testlog.Errorf("unexpected SMP affinity, housekeeping subset: %v, non-housekeeping subset: %v, housekeepCPUSet=%v, minusCpuSet=%v, irqSmpAffinity=%v", isHousekeepingSubset, isMinusSetSubset, housekeepCPUSet, minusCpuSet, irqSmpAffinity)
+	return false, nil
+}
+
+func isIrqBalanceReflectingHousekeeping(ctx context.Context, node *corev1.Node, housekeepCPUSet, containerCpuSet cpuset.CPUSet) (bool, error) {
+	minusCpuSet := containerCpuSet.Difference(housekeepCPUSet)
+	irqBalanceBannedCpus, err := getIrqBalanceBannedCPUs(ctx, node)
+	if err != nil {
+		testlog.Error("failed to get IRQBALANCE_BANNED_CPUS")
+		return false, err
+	}
+	testlog.Infof("IRQBALANCE_BANNED_CPUS: %v", irqBalanceBannedCpus)
+	isHousekeepingSubset := housekeepCPUSet.IsSubsetOf(irqBalanceBannedCpus)
+	isMinusSetSubset := minusCpuSet.IsSubsetOf(irqBalanceBannedCpus)
+	if !isHousekeepingSubset && isMinusSetSubset {
+		return true, nil
+	}
+	testlog.Errorf("unexpected IRQBALANCE_BANNED_CPUS, housekeeping subset: %v, non-housekeeping subset: %v, housekeepCPUSet=%v, minusCpuSet=%v, irqBalanceBannedCpus=%v", isHousekeepingSubset, isMinusSetSubset, housekeepCPUSet, minusCpuSet, irqBalanceBannedCpus)
+	return false, nil
+}
+
+func isHouskeepValidSiblings(ctx context.Context, node *corev1.Node, housekeepCPUSet cpuset.CPUSet) (bool, error) {
+	coreSiblings, err := nodes.GetCoreSiblings(ctx, node)
+	if err != nil {
+		testlog.Errorf("Failed to get core siblings: %v", err)
+		return false, err
+	}
+	for _, numaMap := range coreSiblings {
+		for _, siblings := range numaMap {
+			siblingSet := cpuset.New(siblings...)
+			if housekeepCPUSet.Equals(siblingSet) {
+				return true, nil
+			}
+		}
+	}
+	return false, nil
+}
+
+func getContainerHouskeepCpuSet(testpod *corev1.Pod, containerName string) (cpuset.CPUSet, error) {
+	getHousekeepCpusCmd := []string{"printenv", "OPENSHIFT_HOUSEKEEPING_CPUS"}
+	testpodHousekeepCpusByte, err := pods.ExecCommandOnPod(testclient.K8sClient, testpod, containerName, getHousekeepCpusCmd)
+	if err != nil {
+		testlog.Warningf("error exec pod for housekeeping cpus: %v", err)
+		return cpuset.New(), err
+	}
+	testpodHousekeepCpusStr := strings.TrimSpace(string(testpodHousekeepCpusByte))
+	housekeepingCPUSet, err := cpuset.Parse(testpodHousekeepCpusStr)
+	if err != nil {
+		testlog.Warningf("error parsing housekeeping cpuset: %v", err)
+		return cpuset.New(), err
+	}
+	return housekeepingCPUSet, err
+}
+
+func getContainerCpusSet(testpod *corev1.Pod, containerName string) (cpuset.CPUSet, error) {
```

Comment:
I wonder if we should use the cgroups method and unify this with https://github.com/openshift/cluster-node-tuning-operator/blob/0057ba6284092667b4b80bbe7a3c8d51de9defde/test/e2e/performanceprofile/functests/7_performance_kubelet_node/cgroups.go#L126

Reply:
+1